cmd/notary/tuf.go

```diff
@@ -266,13 +274,108 @@ func verify(cmd *cobra.Command, args []string) {
 	return
 }
 
-func getTransport() *http.Transport {
-	if viper.GetBool("skipTLSVerify") {
-		return &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
+type passwordStore struct {
+	username string
+	password string
+}
+
+func newPasswordStore() auth.CredentialStore {
+	return &passwordStore{
+		username: viper.GetString("username"),
+		password: viper.GetString("password"),
+	}
+}
+
+func (ps *passwordStore) Basic(u *url.URL) (string, string) {
+	var username, password string
+	stdin := bufio.NewReader(os.Stdin)
+	if ps.username != "" {
+		username = ps.username
+		password = ps.password
+	} else {
+		fmt.Fprintf(os.Stdout, "Enter username: ")
```

Comment:
This is not necessary the hub username, this could be basic auth requested directly by a notary server or basic auth requested by a token server. Either way these do not have to be the hub. We could possibly put the server address which the authentication challenge was received from.